spec/unit/connection_spec.rb

```diff
@@ -2,13 +2,40 @@
 
 describe 'Connection' do
   it 'accepts a proc for ldap_config' do
-    ::Devise.ldap_config = Proc.new() {{
+    ::Devise.ldap_config = Proc.new() { {
       'host' => 'localhost',
       'port' => 3389,
       'base' => 'ou=testbase,dc=test,dc=com',
       'attribute' => 'cn',
-    }}
+    } }
     connection = Devise::LDAP::Connection.new()
     expect(connection.ldap.base).to eq('ou=testbase,dc=test,dc=com')
   end
+
+  describe 'When verifying user groups' do
+    before :all do
+      default_devise_settings!
+      reset_ldap_server!
+    end
+    context 'through ldap_ad_group_check' do
+      it 'should search using user_lookup_attribute and group_lookup_attribute', :focus => true do
+        admin = Factory.create(:admin)
+        #
+        ::Devise.ldap_config = Proc.new() {{
+            'host' => 'localhost',
+            'port' => 3389,
+            'base' => 'ou=testbase,dc=test,dc=com',
+            'attribute' => 'cn',
+            'user_lookup_attribute' => 'mail',
+            'group_lookup_attribute' => 'memberof',
+            'admin_user' => 'cn=admin,dc=test,dc=com',
+            'admin_password' => 'secret'
+        }}
+        ::Devise.ldap_ad_group_check = true
+        ::Devise.ldap_auth_username_builder = Proc.new() {|attribute, login, ldap| "#{attribute}=#{login},#{ldap.base}"}
+        connection = Devise::LDAP::Connection.new(:login => admin.email, :password => admin.password, :admin => true)
+        connection.in_group? ''
```

Comment:
we need to do a little more here, our tests should specify a group_name to do a lookup on.

We also need to test the different cases, like checking a user that IS in a particular group, and checking a user that ISN'T in a particular group, etc.